python/ray/data/dataset.py

```diff
@@ -2771,6 +2772,8 @@ def write_parquet(
                 instead of ``arrow_parquet_args`` if any of your write arguments
                 can't pickled, or if you'd like to lazily resolve the write
                 arguments for each dataset block.
+            num_rows_per_file: The number of rows to write to each file. If ``None``,
+                Ray Data writes a system-chosen number of rows to each file.
```

Comment:
> The number of rows to write to each file.
Add a note that this number is not accurate? The actual number of rows will be slightly larger.

> If ``None``,  Ray Data writes a system-chosen number of rows to each file.
We can say the size is based on the max target block size by default.

Reply:
It's typically less, right? Unless each block contains more rows than the number specified?

> We can say the size is based on the max target block size by default.

Not strongly opposed, but this is kinda leaking our implementation? Not sure if we want to expose this as part of the interface.

Reply:
> It's typically less, right? Unless each block contains more rows than the number specified?
why is it less? IIRC, the bundler will accumulate input blocks until the total number of rows is more than the target. 

> Not strongly opposed, but this is kinda leaking our implementation? Not sure if we want to expose this as part of the interface.
agreed not to leak implementation. I felt "system-chosen" was too vague, and wanted to suggest something like "the file size will be based on the upstream outputs" in the beginning, which leaked more details. then I changed to "based on max target size". It's still leaking implementation to some extend though, but doesn't seem too bad because the block size is already a public api. 
I don't have strong preference either. it's fine to keep it as it is.

Reply:
> why is it less? IIRC, the bundler will accumulate input blocks until the total number of rows is more than the target.

Not exactly. If an input block would cause the number of rows to exceed the target, it's excluded.

Here's a simple example: you add two two-row bundles to a bundler with a target of three rows. When you get next bundle, there are two rows instead of four.

```python
import pyarrow as pa

import ray
from ray.data._internal.execution.interfaces import RefBundle
from ray.data._internal.execution.operators.map_operator import _BlockRefBundler
from ray.data.block import BlockAccessor

table = pa.Table.from_pylist([{"spam": 0}, {"spam": 1}])
metadata = BlockAccessor.for_block(table).get_metadata(None, None)
bundle = RefBundle(((ray.put(table), metadata),), owns_blocks=True)


bundler = _BlockRefBundler(3)
bundler.add_bundle(bundle)
bundler.add_bundle(bundle)
assert bundler.has_bundle()
print(bundler.get_next_bundle().num_rows())  # 2
```